src/cpp/src/whisper_pipeline.cpp

```diff
@@ -44,12 +87,13 @@ class WhisperPipeline::WhisperPipelineStatefulImpl : public WhisperPipeline::Whi
 
         m_models.encoder =
             core.compile_model((models_path / "openvino_encoder_model.xml").string(), device, compile_properties)
-                .create_infer_request();
-        m_models.decoder =
-            core.compile_model((models_path / "openvino_decoder_model.xml").string(), device, compile_properties)
-                .create_infer_request();
+                               .create_infer_request();
+        m_models.decoder = core.compile_model((models_path / "openvino_decoder_model.xml").string(), device, compile_properties)
+                               .create_infer_request();
+        auto decoder_with_past_model = core.read_model(models_path / "openvino_decoder_with_past_model.xml");
+        add_attention_mask_input(decoder_with_past_model);
```

Comment:
Is it required to add attention mask for stateful pipeline?

Reply:
No, `stateful` (in fact `dynamic`) pipeline handles it by providing dynamic shape input. 

For static shape pipeline attention mask is needed to specify what part of static shape input should be used.

Reply:
Should it be removed then?

Reply:
These changes are prototype from Sergey, need to revert all changes in [whisper.cpp](https://github.com/openvinotoolkit/openvino.genai/pull/1103/files#diff-9eb3f21f8f27be6218b861f0159dc50546537f1bef39eb1a4ecf9b626ec8918c) and [whisper_pipeline.cpp](https://github.com/openvinotoolkit/openvino.genai/pull/1103/files#diff-501fdc7aca2545fe1133079658cae9efe4ab262da7998a0ea2a811714312b76d)

Reply:
Done